### PR TITLE
Determine windows hardware arch correctly

### PR DESCRIPTION
**Pull request by UtahDave**
### What does this PR do?
Determine windows hardware arch correctly

Also allow for setting path to arbitrary windows python environment

### Previous Behavior
Previously assumed AMD64

### New Behavior
Correctly determine Windows client hardware architecture

### Commits signed with GPG?

Yes

### CODE REVIEW

File: salt/client/ssh/__init__.py [hunk 1]
```diff
@@ -782,6 +783,10 @@ def __init__(
         self.serial = salt.payload.Serial(opts)
         self.wfuncs = salt.loader.ssh_wrapper(opts, None, self.context)
         self.shell = salt.client.ssh.shell.gen_shell(opts, **args)
+        if self.winrm:
+            # Determine if Windows client is x86 or AMD64
+            arch, _, _ = self.shell.exec_cmd('powershell $ENV:PROCESSOR_ARCHITECTURE')
```
Comment (damon-atkins) [inline salt/client/ssh/__init__.py]: Power shell is a very heavy way of doing this.

Here are a few alternatives
```
import platform
platform.machine()
```

What about environment variables.
PROCESSOR_ARCHITECTURE=AMD64
PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 42 Stepping 7, GenuineIntel
PROCESSOR_LEVEL=6
PROCESSOR_REVISION=2a07
ProgramData=C:\ProgramData
ProgramFiles=C:\Program Files
ProgramFiles(x86)=C:\Program Files (x86)
ProgramW6432=C:\Program Files

https://ss64.com/nt/syntax-64bit.html

https://msdn.microsoft.com/en-us/library/aa384274.aspx

Power shell is not a requirement for Salt?

From https://stackoverflow.com/questions/12322308/batch-file-to-check-64bit-or-32bit-os
> PROCESSOR_ARCHITECTURE=x86
> Will appear on Win32, and
> PROCESSOR_ARCHITECTURE=AMD64
> will appear for Win64.
> If you are perversely running the 32-bit cmd.exe process then Windows presents two environment variables:
> PROCESSOR_ARCHITECTURE=x86
> PROCESSOR_ARCHITEW6432=AMD64

**Comment by rallytime** [comment]
re-run py2

File: salt/client/ssh/__init__.py (hunk 1 above)
Comment (UtahDave) [reply salt/client/ssh/__init__.py]: @damon-atkins, thanks for the idea, but in this instance we're running a remote command on a Windows machine that likely doesn't have a python interpreter yet.

**Review by twangboy — APPROVED** [review]

**Comment by UtahDave** [comment]
@rallytime are the py3 test failures related to my code? I don't see how it could be.

**Review by gtmanfred — APPROVED** [review]

**Comment by rallytime** [comment]
@UtahDave Nope! :) There's some flaky issues in the PY3 tests taht we need to clean up.

**Comment by damon-atkins** [comment]
I am surprised you merged it with powershell which can take 5 to 8 seconds to run. When their are sub-second options.

**Comment by damon-atkins** [comment]
64 bit prints AMD64
```
C:\Salt-Dev>python
Python 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import platform
>>> print platform.machine()
AMD64
>>> import os
>>> print os.environ.get('PROCESSOR_ARCHITECTURE')
AMD64
>>>
>>> ^Z
```

32Bit still prints AMD64
```
C:\Salt-Dev>python
Python 3.6.2 (v3.6.2:5fd33b5, Jul  8 2017, 04:14:34) [MSC v.1900 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import platform
>>> print(platform.machine())
AMD64
>>>
```

**Comment by damon-atkins** [comment]
```
c:\windows\system32\cmd.exe /s /c "echo %PROCESSOR_ARCHITECTURE% %PROCESSOR_ARCHITEW6432%"
AMD64 %PROCESSOR_ARCHITEW6432%
c:\windows\SysWOW64\cmd.exe /s /c "echo %PROCESSOR_ARCHITECTURE% %PROCESSOR_ARCHITEW6432%"
x86 AMD64
```
Logic example
```
if  proc_archw6432 = '%PROCESSOR_ARCHITEW6432%' then
    arch=proc_arch  # i.e. %PROCESSOR_ARCHITEW6432% is not set
else
   arch=proc_archw6432  # i.e. %PROCESSOR_ARCHITEW6432% is set 
endif
```

**Comment by damon-atkins** [comment]
@UtahDave  are you sure you want to use powershell (8s) instead of cmd.exe (1s).

**Comment by twangboy** [comment]
@damon-atkins This command isn't run within salt. It is shelled out on a remote system that does not have a Python environment.

**Comment by damon-atkins** [comment]
So CMD.exe will work just the same as PowerShell.exe except a lot faster

**Comment by damon-atkins** [comment]
arch, _, _ = self.shell.exec_cmd('powershell $ENV:PROCESSOR_ARCHITECTURE')
Vs
 arch, _, _ = self.shell.exec_cmd('cmd.exe /s /c "echo %PROCESSOR_ARCHITECTURE%"')

**Pull request merged**